File: SIC_Backend/Repositories/IPlacesRepository.cs
using SIC_Backend.Data.DTOs;
using SIC_Backend.Data.Models;

namespace SIC_Backend.Repositories
{
    public interface IPlacesRepository
    {
        public PlaceDTO GetPlaceById(int id);
        public IEnumerable<PlaceDTO> GetPlacesByUserId(string userId);
        public Task CreatePlaceAsync(CreatePlaceModel model);
        public Task UpdatePlaceAsync(int id, UpdatePlaceModel model);
        public Task DeletePlaceAsync(int id);
    }
}

File: sic_home/lib/repositories/interfaces/i_places_repository.dart
import 'package:sic_home/models/create_place_model.dart';
import 'package:sic_home/models/place.dart';
import 'package:sic_home/models/update_place_model.dart';

abstract class IPlacesRepository {
  Future<Place> getPlaceById(int id);

  Future<List<Place>> getPlacesByUser(String userId);

  Future<void> createPlace(CreatePlaceModel place);

  Future<void> updatePlace(int id, UpdatePlaceModel place);

  Future<void> deletePlace(int id);
}
